Networking/Networking/Remote/GenerativeContentRemote.swift

```diff
@@ -199,74 +199,62 @@ private extension GenerativeContentRemote {
+                }
 
-            return
-                ", categories: Given the list of available categories ```\(categories.map { $0.name }.joined(separator: ", "))```, " +
-                "suggest an array of the best matching categories for this product. You can suggest new categories as well."
-        }()
+                return "Given the list of available tags ```\(tags.map { $0.name }.joined(separator: ", "))```, " +
+                        "suggest an array of the best matching tags for this product. You can suggest new tags as well."
+            }()
 
-        let shippingJson = {
-            let weightJson = {
-                guard let weightUnit else {
-                    return ""
+            let categoriesPrompt: String = {
+                guard !categories.isEmpty else {
+                    return "Suggest an array of the best matching categories for this product."
                 }
 
-                return "weight: Guess and provide only the number in ```\(weightUnit)```"
+                return "Given the list of available categories ```\(categories.map { $0.name }.joined(separator: ", "))```, " +
+                        "suggest an array of the best matching categories for this product. You can suggest new categories as well."
             }()
 
-            let dimensionsJson = {
-                guard let dimensionUnit else {
-                    return ""
+            let shippingPrompt = {
+                var dict = [String: String]()
+                if let weightUnit {
+                    dict["weight"] = "Guess and provide only the number in \(weightUnit)"
                 }
 
-                return
-                    ", " +
-                    "length: Guess and provide only the number in ```\(dimensionUnit)```, " +
-                    "width: Guess and provide only the number in ```\(dimensionUnit)```, " +
-                    "height: Guess and provide only the number in ```\(dimensionUnit)```"
+                if let dimensionUnit {
+                    dict["length"] = "Guess and provide only the number in \(dimensionUnit)"
+                    dict["width"] = "Guess and provide only the number in \(dimensionUnit)"
+                    dict["height"] = "Guess and provide only the number in \(dimensionUnit)"
+                }
+                return dict
             }()
 
-            return "shipping: {" +
-                            weightJson +
-                            dimensionsJson +
-                    "}, "
+            return ["name": "The name of the product, in the ISO language code ```\(language)```",
+                    "description": "Product description of around 100 words long in a ```\(tone)``` tone, "
+                    + "in the ISO language code ```\(language)```",
+                    "short_description": "Product's short description, in the ISO language code ```\(language)```",
```

Comment:
These prompts seem to not be clear enough that sometimes the contents are nested in the language code. How about rewording this:

```suggestion
            return ["name": "The name of the product, written in the language with ISO code ```\(language)```",
                    "description": "Product description of around 100 words long in a ```\(tone)``` tone, "
                    + "written in the language with ISO code ```\(language)```",
                    "short_description": "Product's short description, written in the language with ISO code ```\(language)```",
```

Reply:
I agree. Done in https://github.com/woocommerce/woocommerce-ios/pull/11853/commits/24bbeb33713d3ae1779ced829e523bc6a4b54e43